libpod/runtime.go

```diff
@@ -118,13 +117,6 @@ type Runtime struct {
 	secretsManager *secrets.SecretsManager
 }
 
-func init() {
-	// generateName calls namesgenerator.GetRandomName which the
-	// global RNG from math/rand. Seed it here to make sure we
-	// don't get the same name every time.
-	rand.Seed(time.Now().UnixNano())
-}
-
```

Comment:
Do we now require Go 1.20?  At least `go.mod` says 1.18, I’m not sure what is the authoritative source for that decision.

Without Go 1.20, this could lead to much more frequent collisions, we don’t want that.

Reply:
We don't require it yet, but won't go use the behavior of the compiler-version?  Or does it stick to what's in `go.mod`?

Reply:
I didn’t actually test that, but I think in _almost all_ cases it uses the compiler’s version, without special casing the `go.mod` declaration.

Which means that if this successfully compiles with Go 1.19, it is going to be seeded with 1 and not work the way we need.

---

One possible short-term workaround would be to just silence the linter; seeding the global RNG doesn’t exactly _hurt_.

It’s not nearly as clean as modifying the names generator to add a RNG parameter… and I’ll let actual Podman maintainer decide whether the silencing is good enough.

Reply:
Is there some kind of compiler tag or fancy wizardry that could be used to include `rand.Seed(time.Now().UnixNano())` **only** when compiling with go 1.19 and under?

> it is going to be seeded with 1 and not work the way we need.

Another idea, maybe we could have podman set `$GODEBUG=randautoseed=0` to force the automatic-behavior to do things the "old" way.

Reply:
> Is there some kind of compiler tag or fancy wizardry that could be used to include `rand.Seed(time.Now().UnixNano())` **only** when compiling with go 1.19 and under?

Oh, yes. Great point, I have completely forgotten about that.

“Build tags”, compare https://github.com/containers/storage/blob/main/pkg/archive/archive_19.go / https://github.com/containers/storage/blob/main/pkg/archive/archive_110.go .  (The old one might then still need a lint suppression comment, but it would be clearly justified.)

Reply:
so IIUC, this would look something like:

`libpod/runtime_go_120.go`:
```
//go:build !go1.20
// +build !go1.20

package libpod

func init() {
    rand.Seed(time.Now().UnixNano())
}
```

Would that mean the file is skipped for all versions > 1.20 as well?